scripts/oes-data-migration-scripts/migration_v3.12.x_to_v4.0.x.py

```diff
@@ -1408,43 +1566,43 @@ def addDBVersion():
     isd_admin_password = sys.argv[16]
     sapor_host_url = sys.argv[17]
     audit_service_url = sys.argv[18]
-    redis_host = sys.argv[19]
-    redis_port = sys.argv[20]
-    redis_password = sys.argv[21]
+    spin_db_type = sys.argv[19]        
+    redis_host = sys.argv[20]
+    redis_port = sys.argv[21]
+    redis_password = sys.argv[22]
+
+    spin_db_host = sys.argv[20]
+    spin_db_username = sys.argv[21]
+    spin_db_password = sys.argv[22]
 
     # Establishing the platform db connection
     platform_conn = psycopg2.connect(database=platform_db, user=user_name, password=password, host=platform_host,
                                      port=port)
     print('Opened platform database connection successfully')
 
     # Establishing the oesdb db connection
-    oesdb_conn = psycopg2.connect(database=oes_db, user=user_name, password=password,
-                                  host=oes_host, port=port)
+    oesdb_conn = psycopg2.connect(database=oes_db, user=user_name, password=password, host=oes_host, port=port)
     print("Sapor database connection established successfully")
 
     # Establishing the opsmx db connection
-    autopilot_conn = psycopg2.connect(database=autopilot_db, user=user_name, password=password, host=autopilot_host
-                                      , port=port)
+    autopilot_conn = psycopg2.connect(database=autopilot_db, user=user_name, password=password, host=autopilot_host, port=port)
     print("autopilot database connection established successfully")
 
     # Establishing the audit db connection
     audit_conn = psycopg2.connect(database=audit_db, user=user_name, password=password, host=audit_host, port=port)
     print('Opened audit database connection successfully')
 
     # Establishing the visibility db connection
-    visibility_conn = psycopg2.connect(database=visibility_db, user=user_name, password=password, host=visibility_host,
-                                       port=port)
+    visibility_conn = psycopg2.connect(database=visibility_db, user=user_name, password=password, host=visibility_host, port=port)
     print("Visibility database connection established successfully")
 
-    #Establishing the redis connection
-    redis_conn = redis.Redis(host=redis_host, port=redis_port, password=redis_password)
-    print("Redis connection established successfully")
-
+    
+    redis_conn = get_redis_conn()
```

Comment:
@sudhakaropsmx  similar to my prev comment, you should check the spin_db_type before creating connection otherwise you will end up attempting to create redis connection with incorrect (sql) values and vice versa...we know it will be one at a time so one of this or spindb connection statement (at line 1606) is going to fail (may or may not throw error but doesn't make sense creating connection with incorrect data)

Reply:
oops! I just saw that you are making that check in the get connection methods so it won't fail. Upto you, what I just suggested seems cleaner way to me.

Reply:
nothing change here